core/src/main/java/io/syndesis/core/KeyGenerator.java

```diff
@@ -32,40 +36,43 @@
  */
 public class KeyGenerator {
 
-    private static long lastTimestamp = System.currentTimeMillis();
+    private static AtomicLong lastTimestamp = new AtomicLong(currentTimeMillis());
     private static final byte randomnessByte;
-    private static long randomnessLong;
-
-    private KeyGenerator() {}
+    private static AtomicLong randomnessLong;
 
     static {
-        Random random = new Random();
+        final Random random = ThreadLocalRandom.current();
```

Comment:
Random values are used only on initialization see the [description of the algorithm](https://firebase.googleblog.com/2015/02/the-2120-ways-to-ensure-unique_68.html):

    A push ID contains 120 bits of information. The first 48 bits are a timestamp, which both reduces the chance of collision and allows consecutively created push IDs to sort chronologically.
    The timestamp is followed by 72 bits of randomness, which ensures that even two people creating push IDs at the exact same millisecond are extremely unlikely to generate identical IDs.
    One caveat to the randomness is that in order to preserve chronological ordering if a client creates multiple push IDs in the same millisecond, we just 'increment' the random bits by one.